lega/utils/key.py

```diff
@@ -0,0 +1,159 @@
+import logging
+
+LOG = logging.getLogger(__name__)
+
+class Key():
+
+    def __init__(self, config):
+        """
+        Initiliazing the Key object from a dict
+        """
+        self.config = config
+
+    def public(self):
+        """
+        The public key part
+        Return bytes
+        """
+        raise NotImplementedError('subclasses of Key must provide an public() method')
+
+    def private(self):
+        """
+        The private key part
+        Return bytes
+        """
+        raise NotImplementedError('subclasses of Key must provide an private() method')
+
+
+
+class C4GHFileKey():
```

Comment:
Inherit from baseclass
```suggestion
class C4GHFileKey(Key):
```

Reply:
Yup. my bad.